extensions/ql-vscode/src/query-history/store/query-history-store.ts

```diff
@@ -22,71 +32,54 @@ export async function readQueryHistoryFromFile(
       return [];
     }
 
-    const data = await readFile(fsPath, "utf8");
-    const obj = JSON.parse(data);
-    if (![1, 2].includes(obj.version)) {
+    const obj: QueryHistoryData = await readJson(fsPath, {
+      encoding: "utf8",
+    });
+
+    if (!ALLOWED_QUERY_HISTORY_VERSIONS.includes(obj.version)) {
       void showAndLogExceptionWithTelemetry(
         redactableError`Can't parse query history. Unsupported query history format: v${obj.version}.`,
       );
       return [];
     }
 
     const queries = obj.queries;
-    const parsedQueries = queries
-      // Remove remote queries, which are not supported anymore.
-      .filter((q: QueryHistoryInfo | { t: "remote" }) => q.t !== "remote")
-      .map((q: QueryHistoryInfo) => {
-        // Need to explicitly set prototype since reading in from JSON will not
-        // do this automatically. Note that we can't call the constructor here since
-        // the constructor invokes extra logic that we don't want to do.
-        if (q.t === "local") {
-          Object.setPrototypeOf(q, LocalQueryInfo.prototype);
-
-          // Date instances are serialized as strings. Need to
-          // convert them back to Date instances.
-          (q.initialInfo as any).start = new Date(q.initialInfo.start);
-          if (q.completedQuery) {
-            // Again, need to explicitly set prototypes.
-            Object.setPrototypeOf(
-              q.completedQuery,
-              CompletedQueryInfo.prototype,
-            );
-            Object.setPrototypeOf(
-              q.completedQuery.query,
-              QueryEvaluationInfo.prototype,
-            );
+    // Remove remote queries, which are not supported anymore.
+    const parsedQueries = queries.filter(
+      (q: QueryHistoryDataItem | { t: "remote" }) => q.t !== "remote",
+    );
 
-            // Previously, there was a typo in the completedQuery type. There was a field
-            // `sucessful` and it was renamed to `successful`. We need to handle this case.
-            if ("sucessful" in q.completedQuery) {
-              (q.completedQuery as any).successful = (
-                q.completedQuery as any
-              ).sucessful;
-              delete (q.completedQuery as any).sucessful;
-            }
+    // Map the data models to the domain models.
+    const domainModels: QueryHistoryInfo[] = parsedQueries.map((d) => {
+      if (d.t === "variant-analysis") {
+        const query: VariantAnalysisHistoryItem = d;
```

Comment:
I really like this approach of declaring the type of `query` here, because it means that it enforces that `VariantAnalysisHistoryItem` and `VariantAnalysisDataItem` are consistent. This should stop them diverging before we can implement the data/domain mappers for variant analyses.

We're not breaking the type system here, but we're getting it to work for us to tell us if the types aren't equivalent.